packages/@expo/cli/src/api/rest/wrapFetchWithUserAgent.ts

```diff
@@ -0,0 +1,43 @@
+import process from 'node:process';
+
+import { FetchLike } from './client.types';
+
+export function wrapFetchWithUserAgent(fetch: FetchLike): FetchLike {
+  // Version is added in the build script
+  const userAgent = `expo-cli/${process.env.__EXPO_VERSION}`;
+
+  return (url, init = {}) => {
+    init.headers ??= {};
+
+    if (Array.isArray(init.headers)) {
+      const header = init.headers.find(([name]) => name === 'User-Agent');
+      if (header) {
+        header[1] = `${header[1]}, ${userAgent}`;
+      } else {
+        init.headers.push(['User-Agent', userAgent]);
+      }
+    } else if (canAppendHeader(init.headers)) {
+      init.headers.append('User-Agent', userAgent);
+    } else if (typeof init.headers === 'object') {
+      const headers = init.headers as Record<string, string | string[]>;
+      if (Array.isArray(headers['User-Agent'])) {
+        headers['User-Agent'].push(userAgent);
+      } else if (headers['User-Agent']) {
+        headers['User-Agent'] = `${headers['User-Agent']}, ${userAgent}`;
+      } else {
+        headers['User-Agent'] = userAgent;
+      }
+    }
+
+    return fetch(url, init);
```

Comment:
nit: if we rely on the `fetch` global here, I'd feel it'd be easier to simply instantiate the `new Request` and modify the headers, or to instantiate the `new Headers` and then set `User-Agent` there 🤔 

depends ofc though on whether/how `fetch` is polyfilled, if it is, and hence if `Request` matches up and if it's present

Reply:
The global `fetch` and polyfilled (undici) fetch should be identical, both accepting `headers: new Headers`

We could also just do:
```ts
const headers = new Headers(init.headers);
headers.append('User-Agent', userAgent);
return fetch(URL, { ...init, headers });
```

Reply:
Let me do that instead